app/components/SignupForm.tsx

```diff
@@ -1,34 +1,215 @@
 "use client";
-import React from "react";
+import React, { useEffect, useState } from "react";
+
+interface IUserInput {
+  id: string;
+  name: string;
+  email?: string;
+  password: string;
+  confirmPassword: string;
+}
+
+interface IError {
+  att: string;
+  msg: string;
+}
 
 const SignupForm: React.FC = () => {
+  const [userInput, setUserInput] = useState<IUserInput>({
+    id: "",
+    name: "",
+    password: "",
+    confirmPassword: "",
+  });
+  const [isActive, setIsActive] = useState(false);
+
+  const validateAndSetError = () => {
+    const { id, name, email, password, confirmPassword } = userInput;
+    const emailCheck = new RegExp("[a-z0-9]+@[a-z]+.[a-z]{2,3}");
+    const pwCheck = new RegExp("/^[0-9a-zA-Z]+$/");
+    let isOkay = true;
+
+    if (id.length < 5 || id.length > 15) {
+      setError({
+        att: "id",
+        msg: "ID는 5자 이상 15자 이하로 입력해주세요.",
+      });
+      isOkay = false;
+    } else {
+      deleteError("id");
+    }
+    if (name.length === 0) {
+      setError({
+        att: "name",
+        msg: "이름을 입력해주세요.",
+      });
+      isOkay = false;
+    } else {
+      deleteError("name");
+    }
+    if (email && emailCheck.test(email) === false) {
```

Comment:
사소한 부분이지만 단순화할 수 있습니다.
```suggestion
     if (email && !emailCheck.test(email)) {
```